helpers/deployContracts.ts

```diff
@@ -1,31 +1,277 @@
+
+    const pluginRepoFactoryFactory = new aragonContracts.PluginRepoFactory__factory();
+    const pluginRepoFactory = await pluginRepoFactoryFactory
+      .connect(owner)
+      .deploy(pluginRepoRegistry.address);
+
+    await managingDao.grant(
+      pluginRepoRegistry.address,
+      pluginRepoFactory.address,
+      await pluginRepoRegistry.REGISTER_PLUGIN_REPO_PERMISSION_ID()
+    );
+
+    const pluginSetupProcessorFacotry = new aragonContracts.PluginSetupProcessor__factory();
+    const pluginSetupProcessor = await pluginSetupProcessorFacotry
+      .connect(owner)
+      .deploy(managingDao.address, pluginRepoRegistry.address);
+
+    // dao registry
+    const daoRegistryFactory = new aragonContracts.DAORegistry__factory();
+    const daoRegistry = await daoRegistryFactory.connect(owner).deploy();
+    await daoRegistry.initialize(managingDao.address, daoRegistrar.address);
+    await managingDao.grant(
+      daoRegistrar.address,
+      daoRegistry.address,
+      await daoRegistrar.REGISTER_ENS_SUBDOMAIN_PERMISSION_ID()
+    );
+
+    // dao
+    const daoFactoryFactory = new aragonContracts.DAOFactory__factory();
+    const daoFactory = await daoFactoryFactory
+      .connect(owner)
+      .deploy(daoRegistry.address, pluginSetupProcessor.address);
+    await managingDao.grant(
+      daoRegistry.address,
+      daoFactory.address,
+      await daoRegistry.REGISTER_DAO_PERMISSION_ID()
+    );
+
+    const pluginRepo_Factory = new aragonContracts.PluginRepo__factory();
+
+    const erc20SetupFactory = new aragonContracts.ERC20VotingSetup__factory();
+    const erc20PluginSetup = await erc20SetupFactory.connect(owner).deploy();
+    const erc20RepoAddr = await deployPlugin(
+      pluginRepoFactory,
+      erc20PluginSetup.address,
+      "ERC20Voting",
+      [1, 0, 0],
+      owner
+    );
+    const erc20Repo = pluginRepo_Factory.connect(owner).attach(erc20RepoAddr);
+
+    const allowListFactory = new aragonContracts.AllowlistVotingSetup__factory();
+    const allowListPluginSetup = await allowListFactory.connect(owner).deploy();
+    const allowListRepoAddr = await deployPlugin(
+      pluginRepoFactory,
+      allowListPluginSetup.address,
+      "AllowlistVoting",
+      [1, 0, 0],
+      owner
+    );
+    const allowListRepo = pluginRepo_Factory
+      .connect(owner)
+      .attach(allowListRepoAddr);
+
+    // send ETH to hardcoded wallet in tests
+    await owner.sendTransaction({
+      to: WALLET_ADDRESS,
+      value: ethers.utils.parseEther("50.0"),
+    });
+
+    return {
+      daoFactory,
+      erc20Repo,
+      erc20PluginSetup,
+      allowListRepo,
+      allowListPluginSetup,
+    };
+  } catch (e) {
+    throw e;
+  }
+}
+
+async function deployPlugin(
+  pluginRepoFactory: aragonContracts.PluginRepoFactory,
+  setupAddr: string,
+  name: string,
+  version: [BigNumberish, BigNumberish, BigNumberish],
+  owner: Signer
+) {
+  const repoaddr = await pluginRepoFactory.callStatic.createPluginRepoWithVersion(
```

Comment:
Will the address match 100% with the analogous transaction executed below below?

Reply:
in the testing environment yes. because the transactions are not run in parallel